src/state/nfts/createNftsStore.ts

```diff
@@ -0,0 +1,485 @@
+import { Address } from 'viem';
+import { arcClient } from '@/graphql';
+import { logger, RainbowError } from '@/logger';
+import { createQueryStore } from '@/state/internal/createQueryStore';
+import { time } from '@/utils';
+import { NftsState, NftParams, NftsQueryData, PaginationInfo, Collection } from './types';
+import { parseUniqueAsset, parseUniqueId } from '@/resources/nfts/utils';
+import { useBackendNetworksStore } from '../backendNetworks/backendNetworks';
+import { IS_DEV } from '@/env';
+import { queryClient } from '@/react-query';
+import { showcaseTokensQueryKey } from '@/hooks/useFetchShowcaseTokens';
+import { hiddenTokensQueryKey } from '@/hooks/useFetchHiddenTokens';
+
+export const PAGE_SIZE = 12;
+
+export const mergeMaps = (map1: Map<string, Collection>, map2: Map<string, Collection>) => {
+  return new Map(
+    (function* () {
+      yield* map1;
+      yield* map2;
+    })()
+  );
+};
+
+export const getShowcaseCollectionIds = (address?: string): Set<string> => {
+  const showcaseTokens = queryClient.getQueryData<string[] | undefined>(showcaseTokensQueryKey({ address })) || [];
+  const collectionIds = new Set<string>();
+
+  if (showcaseTokens && Array.isArray(showcaseTokens)) {
+    showcaseTokens.forEach((token: string) => {
+      const parsed = parseUniqueId(token);
+      if (parsed.network && parsed.contractAddress) {
+        const network = replaceEthereumWithMainnet(parsed.network);
+        if (network) {
+          const collectionId = `${network}_${parsed.contractAddress}`.toLowerCase();
+          collectionIds.add(collectionId);
+        }
+      }
+    });
+  }
+
+  return collectionIds;
+};
+
+export const getHiddenCollectionIds = (address?: string): Set<string> => {
+  const hiddenTokens = queryClient.getQueryData<string[] | undefined>(hiddenTokensQueryKey({ address })) || [];
+  const collectionIds = new Set<string>();
+
+  if (hiddenTokens && Array.isArray(hiddenTokens)) {
+    hiddenTokens.forEach((token: string) => {
+      const parsed = parseUniqueId(token);
```

Comment:
seems this internal function is the same as getShowcase? nitty but could pull the function out if so